### PR TITLE
[BE] Added Identity. Refactored User and deleted Role

### PR DESCRIPTION
**Pull request by mislav333**
Integrated IdentityFramework and removed the Role model, as well as it's Dto and Mapper.
Implemented account registration and user management
Adjusted UserRepo methods to suit the new user management

### CODE REVIEW

File: backend/Program.cs [hunk 1]
```diff
@@ -14,6 +20,37 @@
     options.UseNpgsql(builder.Configuration.GetConnectionString("DefaultConnection"));
 });
 
+builder.Services.AddIdentity<User, IdentityRole>(options => {
+    options.Password.RequireDigit = false;
+    options.Password.RequireLowercase = false;
+    options.Password.RequireUppercase = false;
+    options.Password.RequireNonAlphanumeric = false;
+    options.Password.RequiredLength = 6;
```
Comment (LHaler03) [inline backend/Program.cs]: Can be removed since 6 is the default value

File: backend/Dtos/Account/RegisterDto.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        [Required]
+        [EmailAddress]
+        public string? Email { get; set; }
+        [Required]
+        public string? Password { get; set; }
+        [Required]
+        public string? FirstName { get; set; }
+        [Required]
+        public string? LastName { get; set; } 
+
```
Comment (LHaler03) [inline backend/Dtos/Account/RegisterDto.cs]: The empty line can be removed for consistency with other files

File: backend/backend.csproj [hunk 3]
```diff
@@ -7,12 +7,14 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="8.0.10" />
+    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="8.0.10" />
     <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.10" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.10" />
```
Comment (LHaler03) [inline backend/backend.csproj:11]: Are you sure we do not need this package any more?

File: backend/appsettings.json [hunk 4]
```diff
@@ -5,5 +5,10 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "JWT": {
+      "Issuer": "http://localhost:5185",
+      "Audience": "http://localhost:5185",
+      "SigningKey": "mysecret"
```
Comment (LHaler03) [inline backend/appsettings.json:12]: I guess we will replace this in the future before deployment?

File: backend/Controllers/UserController.cs
Comment (LHaler03) [inline backend/Controllers/UserController.cs:1]: Do we need this controller? Can these endpoints be in the `AccountController`?

File: backend/Program.cs (hunk 1 above)
Comment (mislav333) [reply backend/Program.cs]: True, thanks for the heads up! These should also be considered changing later on.

File: backend/Dtos/Account/RegisterDto.cs (hunk 2 above)
Comment (mislav333) [reply backend/Dtos/Account/RegisterDto.cs]: Removed

File: backend/backend.csproj (hunk 3 above)
Comment (mislav333) [reply backend/backend.csproj:11]: I seem to have removed it by accident, added it back.

File: backend/appsettings.json (hunk 4 above)
Comment (mislav333) [reply backend/appsettings.json:12]: Yes, from my understanding the Issuer should be the DNS or IP address of our EC2 instance or our domain, while the Audience is the URL of our API

**Pull request merged**